tree/ntuple/v7/inc/ROOT/RColumnElement.hxx

```diff
@@ -196,6 +199,46 @@ static void CastDeltaSplitUnpack(void *destination, const void *source, std::siz
    }
 }
 
+/// \brief Packing of columns with zigzag + split encoding
+///
+/// Apply split encoding to zigzag-encoded values, used for signed integers
+template <typename DestT, typename SourceT>
+static void CastZigzagSplitPack(void *destination, const void *source, std::size_t count)
+{
+   using UDestT = std::make_unsigned_t<DestT>;
+   constexpr std::size_t kNBitsDestT = sizeof(DestT) * 8;
+   constexpr std::size_t N = sizeof(DestT);
+   auto src = reinterpret_cast<const SourceT *>(source);
+   auto splitArray = reinterpret_cast<char *>(destination);
+   for (std::size_t i = 0; i < count; ++i) {
+      UDestT val = (static_cast<DestT>(src[i]) << 1) ^ (static_cast<DestT>(src[i]) >> (kNBitsDestT - 1));
+      ByteSwapIfNecessary(val);
+      for (std::size_t b = 0; b < N; ++b) {
+         splitArray[b * count + i] = reinterpret_cast<char *>(&val)[b];
+      }
```

Comment:
```suggestion
      auto char_val = reinterpret_cast<char *>(&val);
      for (std::size_t b = 0; b < N; ++b) {
         splitArray[b * count + i] = char_val[b];
      }
```

Reply:
Having conflicting opinions here, @pcanal unless you feel strongly about it I'd leave it as is.

Reply:
I suppose we could even use:
```suggestion
      char const *const auto char_val = reinterpret_cast<char * const>(&val);
      for (std::size_t b = 0; b < N; ++b) {
         splitArray[b * count + i] = char_val[b];
      }
```
'technically' helping the compiler (but it likely can figure it on its own :) ).

The main difference between the original and the alternative is that in the original the emphasis is placed (by the number of characters and potentially code coloring) on the `reinterpret_cast` while the real meat of the line is the indexes (`b * count + i` vs `b`)